src/org/zaproxy/zap/extension/ascan/ActiveScan.java

```diff
@@ -262,12 +263,18 @@ public void notifyNewMessage(final HttpMessage msg) {
 		}
 		
 		this.rcTotals.incResponseCodeCount(msg.getResponseHeader().getStatusCode());
-		
-        if (hRef != null && this.rcTotals.getTotal() <= this.maxResultsToList) {
-            // Very large lists significantly impact the UI responsiveness
-            // limiting them makes large scans _much_ quicker
-        	addHistoryReference(hRef);
-    	}
+
+		if (hRef != null) {
+			// Very large lists significantly impact the UI responsiveness
+			// limiting them makes large scans _much_ quicker
+			if (this.rcTotals.getTotal() <= this.maxResultsToList) {
+				addHistoryReference(hRef);
+			} else {
+				HistoryReference remRef = getMessagesTableModel().getEntry(0).getHistoryReference();
+				removeHistoryReferenceInEdt(remRef);

```

Comment:
If so then should this be `updateTableDataInEdt` or `handleFifoInEdt` or ___________?